lib/Dialect/FIRRTL/FIRRTLDialect.cpp

```diff
@@ -278,6 +294,12 @@ void FIRRTLDialect::printType(Type type, DialectAsmPrinter &os) const {
 Operation *FIRRTLDialect::materializeConstant(OpBuilder &builder,
                                               Attribute value, Type type,
                                               Location loc) {
+  // Boolean constants. Boolean attributes are always a special constant type
+  // like ClockType and ResetType.  Since BoolAttrs are also IntegerAttrs, its
+  // important that this goes first.
+  if (auto attrValue = value.dyn_cast<BoolAttr>())
```

Comment:
This is fine, but you could also check `type.cast<FIRRTLType>().isResetType()` etc.  Both work, your approach is more efficient so staying with it makes sense.